mollie/refunds.go

```diff
@@ -0,0 +1,243 @@
+package mollie
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/google/go-querystring/query"
+)
+
+// Refund describe a refund for a certain payment.
+type Refund struct {
+	Resource         string        `json:"resource,omitempty"`
+	ID               string        `json:"id,omitempty"`
+	Amount           *Amount       `json:"amount,omitempty"`
+	SettlementID     string        `json:"settlementId,omitempty"`
+	SettlementAmount *Amount       `json:"settlementAmount,omitempty"`
+	Description      string        `json:"description,omitempty"`
+	Metadata         interface{}   `json:"metadata,omitempty"`
+	Status           *RefundStatus `json:"status,omitempty"`
+	Lines            *OrderLines   `json:"lines,omitempty"` //TODO: this property should be from order.go
+	PaymentID        string        `json:"paymentId,omitempty"`
+	OrderID          string        `json:"orderId,omitempty"`
+	CreatedAt        *time.Time    `json:"createdAt,omitempty"`
+	Links            *RefundLinks  `json:"_links,omitempty"`
+}
+
+// RefundList describes how a list of refunds will be retrieved by Mollie.
+type RefundList struct {
+	Count    int `json:"count,omitempty"`
+	Embedded struct {
+		Refunds []Refund
+	} `json:"_embedded,omitempty"`
+	Links PaginationLinks `json:"_links,omitempty"`
+}
+
+// RefundStatus describes the status of the refund.
+type RefundStatus string
+
+// Valid refund status.
+const (
+	Queued     RefundStatus = "queued"
+	Pending    RefundStatus = "pending"
+	Processing RefundStatus = "processing"
+	Refunded   RefundStatus = "refunded"
+	Failed     RefundStatus = "failed"
+)
+
+// OrderLines describes an array of order line objects.
+type OrderLines struct {
```

Comment:
Here we are mixing concerns from `orders` and `refunds`, so please move this struct to `orders.go` or the relevant file for orders.

Reply:
Ok, I'll move the code. Has anyone taken #12 ? If not, I can do it too if you want. After I finish this revisions of course.

Reply:
So far nobody is working on that one, so let's get this one merged and then you can take #12 next.